site/_posts/2017-10-24-api-contributors-guide.md

```diff
@@ -0,0 +1,252 @@
+* **:klass:** 
+    * Specifies the base class name of the resources under this collection 
+* **:name:**
+    * Name of the action to be performed 
+* **:identifier:**
+    * Name of the corresponding [product feature](https://github.com/ManageIQ/manageiq/blob/f77c75b4659f97be95f8976ef58abe521085c963/db/fixtures/miq_product_features.yml), which is used to control permissions to resources
+* **:collection_actions:** vs **:resource_actions:**
+    * collection actions correspond to those against the collection `/api/coffee` and resource actions are those corresponding to a single resource `/api/coffee/:c_id`
+ 
+Through the above definition, the following endpoints and actions were created:
+```
+GET /api/coffee
+GET /api/coffee/:c_id
+
+POST /api/coffee
+POST /api/coffee { "action": "edit" }
+POST /api/coffee { "action": "delete" }
+POST /api/coffee/:c_id { "action": "edit" } 
+POST /api/coffee/:c_id { "action": "delete" } 
+
+DELETE /api/coffee/:c_id
+```
+### **Where are the methods?**
+Now that the routes exist, let's take care of the methods. Once the new collection has been added to the `api.yml`, the corresponding controller needs to be [created here](https://github.com/ManageIQ/manageiq-api/tree/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api). The controller will look something like:
+```
+module Api
+  class CoffeeController < BaseController 
+  end 
+end 
+```
+Because the `CoffeeController` inherits from the `BaseController`, a lot of what has been defined in the `api.yml` will already work :sparkles:. The [Base Controller](https://github.com/ManageIQ/manageiq-api/blob/4811ec9fcea2424a585560a761e05d56c53535f8/app/controllers/api/base_controller.rb) holds the action methods (`#show`, `#index`, `#create`, `#update`, `#destroy`) that will be called if they are not overridden in the `CoffeeController`.
+## Custom Resource Actions
+Once basic CRUD is established, what good is seeing a coffee if you can't order one? A custom resource action is the solution for that. In traditional REST, this would be solved via `POST /api/coffee/:c_id/order`. We instead use actions, `POST /api/coffee/:c_id { "action": "order" }`
+ 
+With the new order action, the `:resource_actions:` section will look as follows: 
+```
+        :resource_actions:
+            :get:
+            - :name: read
+              :identifier: coffee_show 
+            :post:
+            - :name: edit
+              :identifier: coffee_edit 
+            - :name: delete
+              :identifier: coffee_delete
+            - :name: order
+              :identifier: coffee_order 
+            :delete:
+            - :name: delete
+              :identifier: coffee_delete
+```
+
+Since ordering a coffee isn't a basic CRUD action handled by the generic methods, that will need to be defined in the `CoffeeController`:
+```ruby
+module Api
+  class CoffeeController < BaseController
+    def order_resource(type, id, data)
```

Comment:
I think it would be nice if here there was a note explaining the naming convention/method signature here, and (possible) where that comes from